File: Java/src/Mersenne_Number.java
/**
 * Mersenne Number Checker
 * 
 * @author Jaysmito Mukherjee 
 * @version 2019-A-000
 */
import java.util.*;
public class Mersenne_Number
{
    public static void main()
    {
        System.out.print("\f");
        Scanner sc=new Scanner(System.in);
        System.out.print("Mersenne Number Checker\n~~~~ ~~~~~~ ~~~~~~~\nEnter the number: ");
        int n=sc.nextInt();
        boolean flag=false;
        for(int i=1;i<=n;i++)
        {
            if(n==(int)(Math.pow(2,i))-1)
            flag=true;
        }
        if(flag)
            System.out.print(n+" is a Mersenne number.");
        else
            System.out.print(n+" is not a Mersenne number.");        
    }
}

File: Python/Mersenne_Number.py
print("Mersenne Number Checker\n")
n = int(input("Enter the number : "))
flag = False
for i in range(0,n) :
	if 2 ** i - 1 == n:
		flag = True
		break
if flag:
	print(f"{n} is a mersenne number.")
else:
	print(f"{n} is not a mersenne number.")